src/index.js

```diff
@@ -1,32 +1,29 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const API = 'https://rickandmortyapi.com/api/character/';
+const xhttp = new XMLHttpRequest();
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const fetchData = (url_api)=> {
+  return new Promise((resolve, reject) => {
+    xhttp.onreadystatechange = function (e) {
+      if (xhttp.readyState === 4 && xhttp.status === 200)
+        resolve(JSON.parse(xhttp.responseText));
+      else
+        reject(new Error('Sorry something went wrong, please try again later.'))
+    };
+    xhttp.open('GET', url_api, true);
+    xhttp.send();
+  });
+};
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
-    }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
+const exec = async () => {
+  try {
+    let res1 =  await fetchData(API);
+    let res2 =  await fetchData(`${API}\\${res1.results[0].id}`);
+    let res3 =  await fetchData(res2.origin.url);
+    console.log(`Personajes: ${res1.info.count}\nPrimer Personaje: ${res2.name}\nDimensión: ${res3['dimension']}`);
+  } catch (e) {
```

Comment:
¿e = a event?

Recuerda que hacemos código para otros humanos, ser más claros nos dara efectividad a leer el código. 

```suggestion
  } catch (error) {
```